src/preload.py

```diff
@@ -427,15 +427,38 @@ def list_images(detect_flasher_type_images):
     return _list_images()
 
 
+def _find_docker_aufs_root(mountpoint):
+    path = os.path.join(mountpoint, "docker", "aufs", "diff")
+    if os.path.exists(path) and os.path.isdir(path):
+        for folder in os.listdir(path):
+            folder_path = os.path.join(path, folder)
+            if os.path.exists(folder_path) and os.path.isdir(folder_path):
+                files = os.listdir(folder_path)
+                if any(x for x in files if not x.startswith(".")):
+                    return folder_path
+
+
+def _find_docker_overlay2_root(mountpoint):
+    path = os.path.join(mountpoint, "docker", "overlay2")
+    if os.path.exists(path) and os.path.isdir(path):
+        for folder in os.listdir(path):
+            folder_path = os.path.join(path, folder, "diff")
+            if os.path.exists(folder_path) and os.path.isdir(folder_path):
+                files = os.listdir(folder_path)
+                if any(x for x in files if not x.startswith(".")):
+                    return folder_path
```

Comment:
As the above two functions pretty much look the same at first glance, can you add some comments in the places they differ, to make it obvious and quicker to read?

Reply:
Comments added

Reply:
Even better, could you gather the repeated code into a worker function and call into that from these two?

Reply:
This will only make it unreadable, let's keep things simple.

Reply:
I factorized the "find a non empty folder" part

Reply:
Oh I didn't see these comments for some reason. Anyway I added some different code below, check it out, if you still feel that it's more unreadable feel free to merge. I personally think it's a bit nicer though.